src/Message/PurchaseRequest.php

```diff
@@ -70,11 +70,14 @@ private function sign($params, $api_key) {
 
 	public function sendData($data)
     {
+		// FIXME: You're not actually sending any data here.
```

Comment:
Thanks for the extensive feedback. I think both gateways (the one I implemented and this here) generate a hash of the data with a secret, that only the application and the payment provider know. The hash is being sent along with the payment data. So tampering with the data is not really possible, unless you know the secret and hashing method.